plugins/modules/pf.py

```diff
@@ -0,0 +1,320 @@
+  community.general.pf:
+    config: /etc/pf.conf
+    state: reloaded
+'''
+
+RETURN = r'''
+state:
+    description: An output of the desired state.
+    returned: success
+    type: str
+    sample: started
+filter:
+    description: An output of the defined rule set filter.
+    returned: success
+    type: str
+    sample: nat
+rule_set:
+    description: An output of the active rule set (only dry-run).
+    returned: success
+    type: str
+    sample: pass in all flags S/SA keep state
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """ start main program to manage packet filter (pf) """
+    module = AnsibleModule(
+        argument_spec=dict(
+            config=dict(required=True, type='path'),
+            filter=dict(default='ignore', choices=['ignore', 'filter', 'nat', 'options']),
+            state=dict(default='reloaded', choices=['started', 'stopped', 'restarted', 'reloaded']),
+            dry_run=dict(default=False, type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    config = module.params['config']
+    filter = module.params['filter']
+    state = module.params['state']
+    dry_run = module.params['dry_run']
+    changed = False
+    out = "No information available for this state."
+
+    # Validate for a supported OS
+    validate(module, config)
+    # Get current status of packet filter (pf)
+    active_pf = status_pf(module)
+
+    # Perform user defined and desired state
+    # Start packet filter (pf) only if it is currently not running
+    if state == 'started':
+        if not active_pf:
+            start_pf(module, active_pf)
+            changed = True
+
+    # Stop packet filter (pf) only it is currently running
+    if state == 'stopped':
+        if active_pf:
+            stop_pf(module, active_pf)
+            changed = True
+
+    # Restart of packet filter (pf) can always be performed
+    if state == 'restarted':
+        restart_pf(module, active_pf)
+        changed = True
+
+    # Reload/Flush the defined rule set for packet filter (pf)
+    # Note: May optionally run in dry run mode and print out
+    #       the rule set in Ansible meta output
+    if state == 'reloaded':
+        out = reload_pf(module, config, dry_run, filter)
+        # Dry run will not modify the system
+        if not dry_run:
+            changed = True
+
+    module.exit_json(
+        changed=changed,
+        meta={
+            "state": state,
+            "filter": filter,
+            "rule_set": out
+        }
+    )
+
+
+def validate(module, config):
+    """ Run basic validations """
+    _validate_os(module)
+    _validate_pf(module)
+    _validate_pf_config(module, config)
+
+
+def _validate_os(module):
+    """ Validate the operating system """
+    rc, out, err = module.run_command(['cat', '/etc/os-release'])
+
+    # Validate for a BSD string in output
+    if 'BSD' not in out:
+        msg_err = 'Error: Unsupported OS. This can only be used on BSD systems.'
```

Comment:
The `fail_json` will incontestably present the message as an error, no need to _say_ it. ;-) 
```suggestion
        msg_err = 'Unsupported OS. This can only be used on BSD systems.'
```